auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/EditAccessGroupsController.kt

```diff
@@ -386,6 +400,21 @@ class EditAccessGroupsController(
     private fun getGroupOrgADName(targetGroupName: String, targetOrganisationCode: String?) =
         LDAPConfig.DATAMART_DELTA_PREFIX + targetGroupName + if (targetOrganisationCode.isNullOrEmpty()) "" else "-$targetOrganisationCode"
 
+    private fun getCommentModifications(
+        currentUser: LdapUser,
+        newComment: String
+    ): Array<ModificationItem> {
```

Comment:
Why is this returning an array of ModificationItem instead of just a nullable ModificationItem? It would simplify this function and also lines 87 to 92.

Line 87 even handles null already, even though this function currently can't return null.

Reply:
the code is using the existing userService to apply the update which is currently wired to take in an array of ModificationItem. If I change my code to use a single item then from what I can see I'd need to modify/create three more functions (updateUser, updateUserOnAD, getAuditData) to take the new parameter, unless there's an easier way. Shall I leave as is or change/create new functions?

Reply:
You could do put it in an array on the line where you call userService, like `userService.updateUser(targetUser, arrayOf(commentModification), session, call)`
Still simplifies this class, e.g. this function doesn't need to look like it can create multiple modification items. 

```
if (newComment.isNotEmpty() && newComment != currentUser.comment) {
       val updatedComment = "${currentUser.comment}\n$newComment"
       return getModificationItem("comment", currentUser.comment, updatedComment)
}
return null
```

Reply:
Changed to the suggested code